src/utils/file-upload-utils.js

```diff
@@ -11,10 +14,28 @@ const ALLOWED_FILE_EXTENSIONS = [
   "bmp",
   "ico",
 ]
+const defaultClient = CloudmersiveVirusApiClient.ApiClient.instance;
 
-const validateAndSanitizeFileUpload = async (data) => {
-  const [schema, content] = data.split(",")
-  const fileBuffer = Buffer.from(content, "base64")
+// Configure API key authorization: Apikey
+var Apikey = defaultClient.authentications['Apikey']
+Apikey.apiKey = CM_API_KEY
+
+const apiInstance = new CloudmersiveVirusApiClient.ScanApi()
+
+const scanFileForVirus = (fileBuffer) => {
+  return new Promise((success, failure) => {
+    apiInstance.scanFile(fileBuffer, function(error, data, response) {
+      if (error) {
+        logger.error('Error when calling cloudmersive API')
+        failure(error)
+      } else {
+        logger.info('Virus Scan API called successfully')
+        success(data)
+      }
+    })
+})}
```

Comment:
we probably wanna check the environment and if it's `dev || test`, just return a clean result to avoid calling the actual endpoint.

also, should we have an abstraction for the initialisation of the client (such as a function) so that we can possibly throw if initialisation failed or return a mock client in `dev` or `test` environments

Reply:
Do we want to avoid calling the third-party API in dev & test environment? 
I assumed at least in dev environment, we should have a working environment where the API call is actually working. So if it's not working, we are able to catch it. I understand that we want to prevent dev env from tempering data store in other envs. But this call is a post only call to Cloudmersive and doesn't really affect the data integrity in our backend. With Cloudmersive premium account, we have 20k call per month and the limit shouldn't be an issue. I think we can consider performing this call in dev & test env. 

For the initialization of` apiInstance`, it's actually a local class instance construction rather than a remote initialisation. Check the reference from Cloudmersive:
`The_Cloudmersive_Virus_Scan_API_lets_you_scan_files_and_content_for_viruses_and_identify_security_issues_with_content_.<br>`
   `* The <code>index</code> module provides access to constructors for all the classes which comprise the public API.`
So in this case, the construction of apiInstance should be a possible failing point, `apiInstance.scanFile` is calling the remote API and could fail, which we are already catching the error in the callback. Hope that makes sense.

Reply:
Sure, @kishore03109 added a similar comment and I have already added the necessary check for the required env (API key), if the env variable is not present, an error will be thrown and the API won't be called (if we call it, it will return 401 unauthorized and it's a waste of time and resource). Please check the fix in fe67cacf0628aea53ec3f4813423e1616f791c2e

Reply:
that works but i think the linting is abit off @__@ do you have autoformat on save configured @QiluXie?

Reply:
Thanks for the help on setting up the auto-format by prettier. The link/format issues are fixed in the latest commit here: d9994ddf40a99b8c5e573a9a49a37c961068424c